ds/dataset.go

```diff
@@ -260,7 +302,8 @@ func (d *Dataset) SetBody(thread *starlark.Thread, _ *starlark.Builtin, args sta
 
 	if raw {
 		if str, ok := data.(starlark.String); ok {
-			d.ds.SetBodyFile(qfs.NewMemfileBytes(fmt.Sprintf("data.%s", df), []byte(string(str))))
+			d.write.SetBodyFile(qfs.NewMemfileBytes(fmt.Sprintf("data.%s", df), []byte(string(str))))
+			d.modBody = true
```

Comment:
How do we want to handle this case?

```python
def transform(ds, ctx):
  ds.set_body(ds.get_body())
```

Assuming our serializers are clean, no bytes have changed. Is the body modified because `set_body` has been called, or is the body _not_ modified because the bytes are the same? Latter i s harder/more computationally expensive, but might be worth it.

It's worth noting that functions in `dsio.SaveDataset` will catch this lack of difference during the save process. The question is more about what the correct behaviour is.

Reply:
I think the correct behavior is to modify the body, even if the bytes stay the same. It's both the easiest implementation, and most consistent with the current descriptions of how `set_body` and `get_body` work. The starlark world is isolated from dsio, and in general shouldn't assume that it needs to respect dsio semantics (something else might be calling it, after all).

Reply:
lovely. If you can add a super quick test that ensures this behaviour, and a comment on the `SetBody` method: `any call to SetBody is considered a modification, even if the body data is the same`, I'd be grateful

Reply:
Done! Added a few tests, and found lots of lots of subtle problems along the way. Great suggestion.